tests/protocol/test_extensions_goal_state.py

```diff
@@ -1,93 +1,43 @@
 # Copyright (c) Microsoft Corporation. All rights reserved.
 # Licensed under the Apache License.
-import json
-import os.path
 import re
 
-from azurelinuxagent.common.protocol.extensions_goal_state import ExtensionsGoalState, GoalStateMismatchError, _CaseFoldedDict
-from azurelinuxagent.common.utils import fileutil
-from tests.protocol.mocks import HttpRequestPredicates
-from tests.tools import AgentTestCase, data_dir
+from azurelinuxagent.common.protocol.extensions_goal_state import ExtensionsGoalState, GoalStateMismatchError
+from tests.protocol.mocks import mockwiredata, mock_wire_protocol
+from tests.tools import AgentTestCase, PropertyMock, patch
 
 
-class ExtensionsGoalStateTestCase(HttpRequestPredicates, AgentTestCase):
+class ExtensionsGoalStateTestCase(AgentTestCase):
     def test_compare_should_report_mismatches_between_extensions_config_and_vm_settings(self):
-        from_vm_settings = ExtensionsGoalState.create_from_vm_settings("123", fileutil.read_file(os.path.join(data_dir, "hostgaplugin/vm_settings.json")))
-        from_extensions_config = ExtensionsGoalState.create_from_extensions_config("123", fileutil.read_file(os.path.join(data_dir, "hostgaplugin/ext_conf.xml")))
-
-        from_vm_settings._required_features = ['FORCE_A_MISMATCH_FEATURE']
-        with self.assertRaisesRegexCM(GoalStateMismatchError, "MultipleExtensionsPerHandler.*!=.*FORCE_A_MISMATCH_FEATURE"):
-            ExtensionsGoalState.compare(from_extensions_config, from_vm_settings)
+        with mock_wire_protocol(mockwiredata.DATA_FILE_VM_SETTINGS) as protocol:
+            from_extensions_config = protocol.client.get_extensions_goal_state()
+            from_vm_settings = protocol.client._extensions_goal_state_from_vm_settings
```

Comment:
Maybe consider adding a new interface to fetch this data instead of reading the private value directly (I understand it would be only used in the tests but still would be more preferable than using the property directly no)?

Reply:
I'll leave it as is. The goal state from vm settings is not part of the public interface (the public interface is actually hiding its existence) and this member variable will go away in a few releases anyway.